src/main/java/com/vmware/vipclient/i18n/messages/service/CacheService.java

```diff
@@ -40,24 +46,25 @@ public CacheService(MessagesDTO dto) {
         }
     }
 
-    public void addCacheOfComponent(Map<String, String> dataMap) {
+    public void addCacheOfComponent(Map<String, String> dataMap, final Map<String, Object> cacheProps) {
         String cacheKey = dto.getCompositStrAsCacheKey();
         Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
         if (c != null) {
-            c.put(cacheKey, dataMap);
+            c.put(cacheKey, dataMap, cacheProps);
         }
     }
 
-    public void updateCacheOfComponent(Map<String, String> dataMap) {
+    public void updateCacheOfComponent(Map<String, String> dataMap, Map<String, Object> cacheProps) {
         String cacheKey = dto.getCompositStrAsCacheKey();
         Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
         if (c != null) {
-            Map<String, String> oldmap = c.get(cacheKey);
+            Map<String, Object> oldmap = c.get(cacheKey);
             if (oldmap == null) {
-                c.put(cacheKey, dataMap);
+                c.put(cacheKey, dataMap, cacheProps);
             } else {
                 oldmap.putAll(dataMap);
-                c.put(cacheKey, oldmap);
+                Map<String, String> cachedMessages = (Map<String, String>) oldmap.get(Cache.MESSAGES);
+                c.put(cacheKey, cachedMessages, cacheProps);
```

Comment:
Actually, I did not change this logic. It has been there since before. I am also wondering why we need to retain the  old values from the oldmap. In any case, I don't want to change the logic in this PR because it is out of this PR's scope.

Reply:
Let's revisit this later.